graphql_compiler/macros/macro_expansion.py

```diff
@@ -125,13 +125,21 @@ def _merge_selection_into_target(target_ast, selection_ast, subclass_sets=None):
 
     Args:
         target_ast: AST at the @macro_edge_target directive
+        type_at_target: str name of the type at the target
         selection_ast: AST to merge inside the target. Required to have a nonempty selection set.
         subclass_sets: optional dict mapping class names to the set of its subclass names
     """
     if selection_ast.selection_set is None or not selection_ast.selection_set.selections:
         raise AssertionError(u'Precondition violated. selection_ast is expected to be nonempty {}'
                              .format(selection_ast))
 
+    # Remove @macro_edge_target directive
+    target_ast.directives = [
+        directive
+        for directive in target_ast.directives
+        if directive.name.value != MacroEdgeTargetDirective.name
+    ]
```

Comment:
moved this from the bottom of the function so I'd be free to change the pointer target_ast